cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/MDSKey.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data2.dataset2.lib.table;
+
+import co.cask.cdap.api.common.Bytes;
+import com.google.common.primitives.Ints;
+import com.google.common.primitives.Longs;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Metadata entry key
+ */
+public final class MDSKey {
+  private final byte[] key;
+
+  public MDSKey(byte[] key) {
+    this.key = key;
+  }
+
+  public byte[] getKey() {
+    return key;
+  }
+
+  /**
+   * Splits the keys into the parts that comprise this.
+   */
+  public Splitter split() {
+    return new Splitter(key);
+  }
+
+  /**
+   * Decodes the keys parts, as specified (int, long, byte[], String)
+   */
+  public static final class Splitter {
+    private final ByteBuffer byteBuffer;
+    private Splitter(byte[] bytes) {
+      byteBuffer = ByteBuffer.wrap(bytes);
+    }
+
+    public int getInt() {
+      return byteBuffer.getInt();
+    }
+
+    public long getLong() {
+      return byteBuffer.getLong();
+    }
+
+    public byte[] getBytes() {
+      int len = byteBuffer.getInt();
```

Comment:
here would be a good place to test that there are enough bytes available... otherwise you may run out of memory if the wrong method is called (for example, getBytes() at the position where there is an int).